python/hail/linalg/rowmatrix.py

```diff
@@ -0,0 +1,251 @@
+
+        Parameters
+        ----------
+        entry_expr: :class:`.Float64Expression`
+            Entry expression for numeric matrix entries.
+
+        Returns
+        -------
+        :class:`RowMatrix`
+        """
+
+        check_entry_indexed('write_from_entry_expr/entry_expr', entry_expr)
+
+        mt = matrix_table_source('write_from_entry_expr/entry_expr', entry_expr)
+
+        #  FIXME: remove once select_entries on a field is free
+        if entry_expr in mt._fields_inverse:
+            field = mt._fields_inverse[entry_expr]
+            jrm = mt._jvds.toRowMatrix(field)
+        else:
+            field = Env.get_uid()
+            jrm = mt.select_entries(**{field: entry_expr})._jvds.toRowMatrix(field)
+
+        return cls(jrm)
+
+    @classmethod
+    @typecheck_method(path=str, partition_size=int)
+    def read_block_matrix(cls, path, partition_size):
+        """Creates a row matrix from a stored block matrix.
+
+        Notes
+        -----
+        The number of partitions in the resulting row matrix will equal
+        the ceiling of ``n_rows / partition_size``.
+
+        Setting the partition size to an exact (rather than approximate)
+        divisor or multiple of the block size will reduce superfluous shuffling
+        of data.
+
+        Warning
+        -------
+        The block matrix must be stored in row-major format, as results from
+        :meth:`.BlockMatrix.write` with ``force_row_major=True`` and from
+        :meth:`.BlockMatrix.write_from_entry_expr`. Otherwise,
+        :meth:`read_block_matrix` will produce an error message.
+
+        Parameters
+        ----------
+        path: :obj:`str`
+            Path to block matrix.
+        partition_size: :obj:`int`
+            Number of rows to group per partition.
+        Returns
+        -------
+        :class:`RowMatrix`
+        """
+        jrm = Env.hail().linalg.RowMatrix.readBlockMatrix(Env.hc()._jhc, path, partition_size)
+
+        return cls(jrm)
+
+    @typecheck_method(path=str,
+                      delimiter=str,
+                      header=nullable(str),
+                      add_index=bool,
+                      parallel=nullable(enumeration('separate_header', 'header_per_shard')),
+                      entries=enumeration('full', 'lower', 'strict_lower', 'upper', 'strict_upper'))
+    def export(self, path, delimiter='\t', header=None, add_index=False, parallel=None, entries='full'):
+        """Exports row matrix as a delimited text file.
+
+        Examples
+        --------
+
+        >>> import numpy as np
+        >>> from hail.linalg import BlockMatrix
+        >>> from hail.linalg import RowMatrix
+        >>> nd = np.array([[1.0, 0.8, 0.7],
+        ...                [0.8, 1.0 ,0.3],
+        ...                [0.7, 0.3, 1.0]])
+        >>> BlockMatrix.from_numpy(nd).write('output/bm', force_row_major=True)
+        >>> rm = RowMatrix.read_block_matrix('output/bm', partition_size=2)
```

Comment:
I think this block of code should go in the class description. That way you can use `rm` in all of your examples.